src/sdk/pynni/nni/compression/torch/pruning/structured_pruning.py

```diff
@@ -119,30 +345,56 @@ class L1FilterPrunerMasker(StructuredWeightMasker):
     https://arxiv.org/abs/1608.08710
     """
 
-    def get_mask(self, base_mask, weight, num_prune, wrapper, wrapper_idx):
+    def get_mask(self, base_mask, weight, num_prune, wrapper, wrapper_idx, channel_masks=None):
+        # get the l1-norm sum for each filter
+        w_abs_structured = self.get_channel_sum(wrapper, wrapper_idx)
+        if channel_masks is not None:
+            # if we need to mask some channels in advance
+            w_abs_structured = w_abs_structured * channel_masks
+        threshold = torch.topk(w_abs_structured.view(-1),
+                               num_prune, largest=False)[0].max()
+        mask_weight = torch.gt(w_abs_structured, threshold)[
+            :, None, None, None].expand_as(weight).type_as(weight)
+        mask_bias = torch.gt(w_abs_structured, threshold).type_as(
+            weight).detach() if base_mask['bias_mask'] is not None else None
+
+        return {'weight_mask': mask_weight.detach(), 'bias_mask': mask_bias}
+
+    def get_channel_sum(self, wrapper, wrapper_idx):
+        weight = wrapper.module.weight.data
         filters = weight.shape[0]
         w_abs = weight.abs()
         w_abs_structured = w_abs.view(filters, -1).sum(dim=1)
-        threshold = torch.topk(w_abs_structured.view(-1), num_prune, largest=False)[0].max()
-        mask_weight = torch.gt(w_abs_structured, threshold)[:, None, None, None].expand_as(weight).type_as(weight)
-        mask_bias = torch.gt(w_abs_structured, threshold).type_as(weight).detach() if base_mask['bias_mask'] is not None else None
+        return w_abs_structured
```

Comment:
here we only support output channel, right?

Reply:
Yes, the `L1FilterPruner` only prunes the filters (output channel).